omniqueue/src/backends/redis/streams.rs

```diff
@@ -24,6 +26,15 @@ const LISTEN_STREAM_ID: &str = ">";
 // FIXME(onelson): expose in config?
 const PENDING_BATCH_SIZE: usize = 1000;
 
+macro_rules! internal_to_stream_payload {
+    (($payload:expr, $num_receives:expr), $payload_key:expr) => {
+        &[
+            ($payload_key, $payload),
+            (NUM_RECEIVES, $num_receives.to_string().as_bytes()),
+        ]
+    };
+}
```

Comment:
Why is this a macro, rather than a function?

Reply:
Make it generic over `T: ToString`?

Reply:
Are you essentially suggesting to pass a `String` instead of a `usize`?

Reply:
Err, I think I misunderstood you. Let's just keep it a macro for now ^^

Reply:
I feel like there's got to be something better here and I'm missing something obvious (again, so that the second elements of each tuple will be &[u8]), but I've given up for now.